polaris/polaris/sep10auth/views.py

```diff
@@ -29,150 +35,179 @@
-    source_account = transaction.operations[0].source
-    logger.info(
-        f"Challenge verified, generating SEP-10 token for account {source_account}"
-    )
-    hash_hex = binascii.hexlify(transaction_envelope.hash()).decode()
-    jwt_dict = {
-        "iss": request.build_absolute_uri("/auth"),
-        "sub": transaction.operations[0].source,
-        "iat": issued_at,
-        "exp": issued_at + 24 * 60 * 60,
-        "jti": hash_hex,
-    }
-    encoded_jwt = jwt.encode(jwt_dict, settings.SERVER_JWT_KEY, algorithm="HS256")
-    return encoded_jwt.decode("ascii")
-
-
-def _get_auth(request):
-    account = request.GET.get("account")
-    if not account:
-        return JsonResponse(
-            {"error": "no 'account' provided"}, status=status.HTTP_400_BAD_REQUEST
-        )
+    ###############
+    # GET functions
+    ###############
+    def get(self, request, *args, **kwargs) -> JsonResponse:
+        account = request.GET.get("account")
+        if not account:
+            return JsonResponse(
+                {"error": "no 'account' provided"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            transaction = self._challenge_transaction(account)
+        except Ed25519PublicKeyInvalidError as e:
+            return JsonResponse({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
 
-    try:
-        transaction = _challenge_transaction(account)
         logger.info(f"Returning SEP-10 challenge for account {account}")
-    except Ed25519PublicKeyInvalidError as e:
-        return JsonResponse({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
-
-    return JsonResponse(
-        {
-            "transaction": transaction,
-            "network_passphrase": settings.STELLAR_NETWORK_PASSPHRASE,
-        }
-    )
-
-
-def _post_auth(request):
-    success, xdr_or_error = _transaction_from_post_request(request)
-    if not success:
-        return JsonResponse({"error": xdr_or_error}, status=status.HTTP_400_BAD_REQUEST)
-    envelope_xdr = xdr_or_error
-    validate_error = _validate_envelope_xdr(envelope_xdr)
-    if validate_error != "":
         return JsonResponse(
-            {"error": validate_error}, status=status.HTTP_400_BAD_REQUEST
+            {
+                "transaction": transaction,
+                "network_passphrase": settings.STELLAR_NETWORK_PASSPHRASE,
+            }
         )
-    return JsonResponse({"token": _generate_jwt(request, envelope_xdr)})
 
+    @staticmethod
+    def _challenge_transaction(client_account):
+        """
+        Generate the challenge transaction for a client account.
+        This is used in `GET <auth>`, as per SEP 10.
+        Returns the XDR encoding of that transaction.
+        """
+        return build_challenge_transaction(
+            server_secret=settings.SIGNING_SEED,
+            client_account_id=client_account,
+            anchor_name=ANCHOR_NAME,
+            network_passphrase=settings.STELLAR_NETWORK_PASSPHRASE,
```

Comment:
The Python SDK still has a default timeout of 300ms rather than 900ms as in v1.3.0. I think we should set it explicitly here though so we're making a conscious decision that's what we want the timeout to be. Could we add that?